Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithm.H

```diff
@@ -64,14 +77,33 @@ class PsatdAlgorithm : public SpectralBaseAlgorithm
          * \param[in,out] current    Array of unique pointers to \c MultiFab storing
          *                           the three components of the current density
          */
-        virtual void VayDeposition (SpectralFieldData& field_data,
-                                    std::array<std::unique_ptr<amrex::MultiFab>,3>& current) override final;
+        virtual void VayDeposition (
+            SpectralFieldData& field_data,
+            std::array<std::unique_ptr<amrex::MultiFab>,3>& current) override final;
 
     private:
-        SpectralRealCoefficients C_coef, S_ck_coef, X1_coef, X2_coef, X3_coef;
+
+        // These real and complex coefficients are always allocated
+        SpectralRealCoefficients C_coef, S_ck_coef;
+        SpectralComplexCoefficients T2_coef, X1_coef, X2_coef, X3_coef, X4_coef;
+
+        // These real and complex coefficients are allocated only with averaged Galilean PSATD
+        SpectralRealCoefficients C1_coef, C3_coef, S1_coef, S3_coef;
+        SpectralComplexCoefficients Psi1_coef, Psi2_coef, Psi3_coef, Psi4_coef,
+                                    A1_coef, A2_coef, Rhoold_coef, Rhonew_coef, Jcoef_coef;
+
+        // Centered modified finite-order k vectors
```

Comment:
This would be possible for what concerns the update equations and I would also prefer that. However, the centered vectors could be used in the function `CurrentCorrection` (to compute the dot product with the Galilean velocity), which would be called at each PIC iteration. It is true that we do not frequently use this option, but I'm not sure about how to proceed. What do you think?

Reply:
Ah, yes, sorry: I forgot about the `CurrentCorrection`. Then I agree that we can keep the centered vectors here.